docs/errors.md

```diff
@@ -165,12 +166,58 @@ Check the supported types for encrypted columns.
 
 
 
+<!-- ---------------------------------------------------------------------------------------------------- -->
+
+
+## Statement could not be type checked <a id='mapping-statement-could-not-be-type-checked'></a>
+
+An error occurred when attempting to type check the SQL statement.
+
+### Error message
+
+```
+Statement could not be type checked: '{type-check-erro-message}'
+```
+
+### Notes
+
+CipherStash Proxy checks SQL statements against the database schema to transparently encrypt and decrypt data.
+
+The behaviour of Proxy depends on the `mapping_errors_enabled` configuration.
+
+When `mapping_errors_enabled == false` (the default) type check errors are logged, and the statement is passed through to the database.
+
+When `mapping_errors_enabled == true` type check errors are raised and statement execution halts.
+
+In our experience, most production systems have a relatively small number of columns that require protection.
+As SQL is large and complex, rather than block statements with false negative type check errors, the default behaviour is to allow the statement.
+
+However, this does mean it is possible that a statement referencing encrypted columns cannot be type-checked and is passed through to the database.
```

Comment:
```suggestion
However, this does mean it is possible that a statement that references encrypted columns cannot be type-checked, and it will be passed through to the database.
```